### PR TITLE
Adding tests and Documentation

### PR DESCRIPTION
**Pull request by vdantu**
Before or while filing an issue please feel free to join our [ slack channel](https://join.slack.com/t/mms-awslabs/shared_invite/enQtNDk4MTgzNDc5NzE4LTBkYTAwMjBjMTVmZTdkODRmYTZkNjdjZGYxZDI0ODhiZDdlM2Y0ZGJiZTczMGY3Njc4MmM3OTQ0OWI2ZDMyNGQ) to get in touch with development team, ask questions, find out what's cooking and more!

## Issue #, if available:
Adding tests and documents for the SDK
## Testing done:

**To run CI tests on your changes refer [README.md](https://github.com/awslabs/mxnet-model-server/blob/master/ci/README.md)**

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: serving-sdk/src/test/java/software/amazon/ai/mms/servingsdk/ModelServerEndpointTest.java [hunk 1]
```diff
@@ -13,39 +13,115 @@
 
 package software.amazon.ai.mms.servingsdk;
 
-import junit.framework.Test;
-import junit.framework.TestCase;
-import junit.framework.TestSuite;
+
+import org.junit.Assert;
```
Comment (frankfliu) [inline serving-sdk/src/test/java/software/amazon/ai/mms/servingsdk/ModelServerEndpointTest.java]: still from junit?
Should this be testng?

File: serving-sdk/src/test/java/software/amazon/ai/mms/servingsdk/impl/ModelImpl.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+package software.amazon.ai.mms.servingsdk.impl;
+
+import software.amazon.ai.mms.servingsdk.Model;
+import software.amazon.ai.mms.servingsdk.Worker;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ModelImpl implements Model {
```
Comment (frankfliu) [inline serving-sdk/src/test/java/software/amazon/ai/mms/servingsdk/impl/ModelImpl.java]: Better call it MockModel if you prefer write your own Mock class.

File: serving-sdk/src/test/java/software/amazon/ai/mms/servingsdk/impl/ContextImpl.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+package software.amazon.ai.mms.servingsdk.impl;
+
+import software.amazon.ai.mms.servingsdk.Context;
+import software.amazon.ai.mms.servingsdk.Model;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+
+public class ContextImpl implements Context {
```
Comment (frankfliu) [inline serving-sdk/src/test/java/software/amazon/ai/mms/servingsdk/impl/ContextImpl.java]: Can you use Mockito or PowerMock?

File: serving-sdk/src/test/java/software/amazon/ai/mms/servingsdk/impl/ModelImpl.java (hunk 2 above)
Comment (vdantu) [reply serving-sdk/src/test/java/software/amazon/ai/mms/servingsdk/impl/ModelImpl.java]: Makes sense.

File: serving-sdk/src/test/java/software/amazon/ai/mms/servingsdk/ModelServerEndpointTest.java (hunk 1 above)
Comment (vdantu) [reply serving-sdk/src/test/java/software/amazon/ai/mms/servingsdk/ModelServerEndpointTest.java]: I moved to junit completely.

File: serving-sdk/src/test/java/software/amazon/ai/mms/servingsdk/impl/ModelImpl.java (hunk 2 above)
Comment (vdantu) [reply serving-sdk/src/test/java/software/amazon/ai/mms/servingsdk/impl/ModelImpl.java]: Just documenting our discussion. The initial thought here was to make sure that we have both positive and negative cases. I modified these tests to have Mock and Reflections in the tests instead now.

File: serving-sdk/src/test/java/software/amazon/ai/mms/servingsdk/ModelServerEndpointTest.java [hunk 4]
```diff
@@ -13,39 +13,186 @@
...
+        Mockito.when(m.getModelHandler()).thenReturn("mxnet_service:handle");
+        Mockito.when(m.getModelUrl()).thenReturn("/tmp/model/squeezenet.mar");
+        Mockito.when(m.getModelName()).thenReturn("squeezenet");
+        Mockito.when(w.getWorkerMemory()).thenReturn((long)100);
+        Mockito.when(w.isRunning()).thenReturn(false);
+        Mockito.when(ea.urlPattern()).thenReturn("myEndpoint");
+        Mockito.when(ea.description()).thenReturn("This is a test endpoint");
+        Mockito.when(ea.endpointType()).thenReturn(EndpointTypes.INFERENCE);
+        Mockito.when(rsp.getOutputStream()).thenReturn(outputStream);
+
```
Comment (frankfliu) [inline serving-sdk/src/test/java/software/amazon/ai/mms/servingsdk/ModelServerEndpointTest.java]: Did you format the code?

Comment (vdantu) [reply serving-sdk/src/test/java/software/amazon/ai/mms/servingsdk/ModelServerEndpointTest.java]: Hmm. Today we don't have a auto formatter for maven. But, the checker that we have with maven doesn't seem to throw any warnings.

File: docs/images/helpers/plugins_sdk_class_uml_diagrams [hunk 5]
```diff
@@ -0,0 +1,66 @@
+
```
Comment (frankfliu) [inline docs/images/helpers/plugins_sdk_class_uml_diagrams]: This file should have a extension: .puml

File: docs/mms_endpoint_plugins.md [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+        r.put("BATCH_STRATEGY", "SINGLE_RECORD");
+        r.put("MAX_PAYLOAD_IN_MB", prop.getProperty("max_request_size"));
+        r.put("BATCH", "true");
+        rsp.getOutputStream()
+                .write(
+                        new GsonBuilder()
+                                .setPrettyPrinting()
+                                .create()
+                                .toJson(r)
+                                .getBytes(Charset.forName("UTF-8")));
```
Comment (frankfliu) [inline docs/mms_endpoint_plugins.md]:
```suggestion
                                .getBytes(StandardCharsets.UTF_8));
```

File: docs/images/helpers/plugins_sdk_class_uml_diagrams (hunk 5 above)
Comment (vdantu) [reply docs/images/helpers/plugins_sdk_class_uml_diagrams]: Hmm. I don't think this is required for functionality. But, I can change it for making it more clear without opening the file.

**Pull request merged**